staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -184,8 +199,38 @@ func (c *AvailableConditionController) sync(key string) error {
 			return err
 		}
 	}
-
-	// TODO actually try to hit the discovery endpoint
+	// actually try to hit the discovery endpoint when it isn't local and when we're routing as a service.
+	// TODO figure out if the kube-proxy code handling has similar problems
+	if apiService.Spec.Service != nil && !c.enableAggregatorRouting && c.serviceResolver != nil {
+		url, err := c.serviceResolver.ResolveEndpoint(apiService.Spec.Service.Namespace, apiService.Spec.Service.Name)
+		if err != nil {
+			return err
+		}
+		// construct an http client that will ignore TLS verification (if someone owns the network and messes with your status
+		// that's not so bad) and sets a very short timeout for responsive-ness  Since we requeue on endpoint changes
+		// we'll get kicked as these change.
+		httpClient := &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			},
+		}
+		req, err := http.NewRequest("GET", url.String(), nil)
+		if err != nil {
+			return err
+		}
+		ctx, _ := context.WithTimeout(req.Context(), 5*time.Second)
+		req.WithContext(ctx)
+		resp, err := httpClient.Do(req)
+		if err != nil {
+			availableCondition.Status = apiregistration.ConditionFalse
```

Comment:
> If we just return the httpClient.Do err to requeue, wouldn't the throttling do what we want?

It would end up delaying longer than the resync time, but if you wish, sure.